test/node/index.js

```diff
@@ -89,4 +89,57 @@ describe('parser', function() {
         });
     });
   });
+
+  it('should encode/decode an ArrayBuffer as b64', function(done) {
+    var buffer = new ArrayBuffer(4);
+    var dataview = new DataView(buffer);
+    for (var i = 0; i < buffer.byteLength ; i++) dataview.setInt8(i, i);
+
+    encode({ type: 'message', data: buffer }, function(encoded) {
+      var decoded = decode(encoded, 'arraybuffer');
+      expect(decoded).to.eql({ type: 'message', data: buffer });
+      expect(equals(buffer, decoded.data)).to.be(true);
+      done();
+    });
+  });
+
+  it('should encode/decode an ArrayBuffer as binary', function(done) {
+    var buffer = new ArrayBuffer(4);
+    var dataview = new DataView(buffer);
+    for (var i = 0; i < buffer.byteLength ; i++) dataview.setInt8(i, i);
+
+    encode({ type: 'message', data: buffer }, true, function(encoded) {
+      var decoded = decode(encoded, 'arraybuffer');
+      expect(decoded).to.eql({ type: 'message', data: buffer });
+      expect(equals(buffer, decoded.data)).to.be(true);
+      done();
+    });
+  });
+
+  it('should encode/decode a typed array as binary', function(done) {
+    var buffer = new ArrayBuffer(10);
+    var typedArray = new Int8Array(buffer, 2, 4);
+    for (var i = 0; i < buffer.byteLength ; i++) typedArray[i] = (i + 1);
+
+    encode({ type: 'message', data: typedArray }, true, function(encoded) {
+      var decoded = decode(encoded, 'arraybuffer');
+      expect(decoded).to.eql({ type: 'message', data: buffer });
+      expect(equals(buffer, decoded.data)).to.be(true);
+      done();
+    });
+  });
+
 });
+
+/**
+ * Check whether two ArrayBuffer are equal
+ */
+function equals (buffer1, buffer2) {
```

Comment:
Or we can just use `expect(new Uint8Array(buffer1)).to.eql(new Uint8Array(buffer2));` instead.